packages/loader/container-utils/src/liveSummarySanitizer.ts

```diff
@@ -0,0 +1,194 @@
+            tree: {
+                [".channels"]: parentTree[".channels"],
+                [".electedSummarizer"]: createBlobItem({
+                    electionSequenceNumber: 0,
+                }),
+                [".metadata"]: createBlobItem({
+                    createContainerRuntimeVersion:
+                        meta.createContainerRuntimeVersion,
+                    createContainerTimestamp: meta.createContainerTimestamp,
+                    summaryFormatVersion: meta.summaryFormatVersion,
+                }),
+            },
+        };
+    }
+
+    /**
+     * Sanitize protocol tree
+     * @param srcSummary - summary to transform
+     */
+    private sanitizeProtocolTree(srcSummary: ISummaryTree): ISummaryTree {
+        const protocolSummary = srcSummary.tree[".protocol"];
+        assert(
+            protocolSummary.type === SummaryType.Tree,
+            "Invalid object type",
+        );
+
+        const keys = [
+            "attributes",
+            "quorumMembers",
+            "quorumProposals",
+            "quorumValues",
+        ];
+
+        assert(
+            validateKeyPresence(keys, protocolSummary),
+            "Valid protocol tree keys should be present",
+        );
+
+        const parentTree = protocolSummary.tree;
+        const quorumValues = readBlobContent(
+            (parentTree.quorumValues as ISummaryBlob).content,
+        );
+
+        assert(Array.isArray(quorumValues), "Invalid quorum values");
+        const transformedQuorumValues: [string, ICommittedProposal][] = [];
+        for (const quorumValue of quorumValues) {
+            assert(
+                quorumValue !== undefined && quorumValue.length >= 2,
+                "Quorum value not valid",
+            );
+
+            const codeProposal = quorumValue[this.quorumValIdx];
+            const committedCodeProposal = {
+                key: codeProposal.key,
+                value: codeProposal.value,
+                approvalSequenceNumber: 0,
+                commitSequenceNumber: 0,
+                sequenceNumber: 0,
+            };
+            transformedQuorumValues.push([
+                quorumValue[this.quorumKeyIdx],
+                committedCodeProposal,
+            ]);
+        }
+
+        return {
+            type: SummaryType.Tree,
+            tree: {
+                attributes: createBlobItem({
+                    sequenceNumber: 0,
+                    term: 1,
+                    minimumSequenceNumber: 0,
+                }),
+                quorumMembers: { ...this.emptySummaryBlob },
+                quorumProposals: { ...this.emptySummaryBlob },
+                quorumValues: createBlobItem(transformedQuorumValues),
+            },
+        };
+    }
+}
+
+/**
+ * Extracts ISummaryTree retreived from attached document that can be used to hydrate new (detached) container
+ * @param liveSummary - summary
+ */
+export function getSanitizedCopy(liveSummary: ISummaryTree): ISummaryTree {
```

Comment:
liveSummary is ambiguous and doesn't really convey anything. Maybe downloadedSummary?

Reply:
`getSanitizedCopy` was removed. We are moving to option (3) as discussed here: https://github.com/microsoft/FluidFramework/issues/9442.